msal/authority.py

```diff
@@ -53,15 +53,20 @@ def __init__(self, authority_url, http_client, validate_authority=True):
             performed.
         """
         self._http_client = http_client
-        authority, self.instance, tenant = canonicalize(authority_url)
-        parts = authority.path.split('/')
-        is_b2c = any(self.instance.endswith("." + d) for d in WELL_KNOWN_B2C_HOSTS) or (
+        self.authority, self.instance, self.tenant = canonicalize(authority_url)
```

Comment:
I haven't tried this, but it seems to be a possible direction to explore. What if we just record the 3 raw input here as internal members:

```python
self._http_client = http_client
self._authority_url = authority_url
self._validate_authority = validate_authority
self._is_initialized = False
```

Ant then move ALL the initialization into `initialize()`? That way, the change would likely be cleaner.

Reply:
Getting close. :-)

Suggest to give all these internal properties a leading underscore. That is the only conventional approach to scare downstream app developers away from making a direct dependencies of those properties.